lib/gateways/sessions.rb

```diff
@@ -28,6 +28,10 @@ def search(username: nil, ips: nil)
       end
     end
 
+    def count_distinct_users
+      Session.where(siteIP: organisation_ips).where('start >= ?', 1.day.ago).where(success: true).distinct.count(:username)
```

Comment:
I find it a bit odd that this `UseCase` is used for searching through the past two weeks of logs, and at the same time it returns the number of unique connections in the past day. 

For instance it could return 500 results, as well as a `count_distinct_users` of `0`. This is pretty confusing. I was pondering wether this `count_distinct_users` should be in a different use case and couldn't really make up my mind, but this peculiarity makes me think it should be in a different use case. That way you can also get rid of the `ips:` and `username:` arguments which are superfluous in:
```
@connections = get_auth_requests.execute(
      ips: nil,
      username: nil
    ).fetch(:connection_count)
```

So you could have something like:
```
@connections = get_unique_user_requests_count.execute[:result]
```

Eventually we might end up extending it to take a date range, something like:
```
@connections = get_unique_user_requests_count.execute(date_range: 15.days)[:result]
```

Reply:
I would strongly vote for this UseCase being split up, if time allows.